kratos/containers/pointer_vector_set.h

```diff
@@ -1082,34 +1093,66 @@ class PointerVectorSet final
 
             if (lower_bound_first == lower_bound_last &&
                 lower_bound_first == upper_bound_first &&
-                lower_bound_first == upper_bound_last)
-            {
+                lower_bound_first == upper_bound_last) {
                 // all 4 bounds are equal, hence this can be inserted without checking further
-                mData.reserve(mData.size() + std::distance(first, last));
                 if (lower_bound_first == mData.end()) {
+                    // all bounds are pointing to the end of the vector, hence pushing back.
+                    mData.reserve(mData.size() + std::distance(first, last));
                     for (auto it = first; it != last; ++it) {
                         mData.push_back(TPointerType(&GetReference(it)));
                     }
                 } else {
-                    // now if the capacity of the new mData is larger than the existing
-                    // capacity, then the current lower_bound_first is invalidated.
-                    // hence needs to find it again.
-                    const auto new_lower_bound = std::lower_bound(mData.begin(), mData.end(), KeyOf(GetReference(first)), CompareKey());
-                    auto current_pos = new_lower_bound - 1;
+                    // all bounds are pointing to a middle section, hence creating a new vector
+                    // and will push back to it.
+                    TContainerType temp;
+                    temp.reserve(mData.size() + std::distance(first, last));
+
+                    // insert the original block first
+                    for (auto it = mData.begin(); it != lower_bound_first; ++it) {
+                        temp.push_back(TPointerType(&GetReference(it)));
+                    }
+
+                    // now insert the new items
                     for (auto it = first; it != last; ++it) {
-                        current_pos = mData.insert(current_pos + 1, TPointerType(&GetReference(it)));
+                        temp.push_back(TPointerType(&GetReference(it)));
                     }
+
+                    // now add the rest of the items from the original vector
```

Comment:
I don't have strong comments, just to start comments with capital letter please.